src/Hybrid/transitions.jl

```diff
@@ -0,0 +1,179 @@
+
+    clist_G = constraints_list(tr.G)
+    clist_I⁻ = constraints_list(tr.I⁻)
+    clist_I⁺ = constraints_list(tr.I⁺)
+    clist_X = constraints_list(X)
+    out = vcat(clist_G, clist_I⁻, clist_I⁺, clist_X)
+    flag = remove_redundant_constraints!(out)
+    if flag
+        return HPolyhedron(out)
+    else
+        return EmptySet(dim(X))
+    end
+end
+
+# the source invariant is the set union of half-spaces, the other sets are polyhedral
+function _apply(tr::DiscreteTransition{<:Universe, <:ZeroSet, GT, IT⁻, IT⁺},
+                X::AbstractPolyhedron{N},
+                ::Exact) where {N,
+                                GT<:AbstractPolyhedron{N},
+                                VN,
+                                IT⁻<:UnionSetArray{N, LinearConstraint{N, VN}},
+                                IT⁺<:AbstractPolyhedron{N}}
+
+    clist_G = constraints_list(tr.G)
+    clist_I⁺ = constraints_list(tr.I⁺)
+    clist_X = constraints_list(X)
+    aux = vcat(clist_G, clist_I⁺, clist_X)
+    flag = remove_redundant_constraints!(aux)
+    !flag && return EmptySet(dim(X))
+
+    clist_I⁻ = tr.I⁻.array
+    out = Vector{HPolyhedron{N, Vector{N}}}()
+
+    for ci in clist_I⁻
+        aux2 = vcat(aux, ci)
+        flag = remove_redundant_constraints!(aux2)
+        if !flag
+            continue
+        else
+            push!(out, HPolyhedron(copy(aux2)))
+        end
+    end
+
+    if isempty(out)
+        return EmptySet(dim(X))
+    else
+        return UnionSetArray(out)
+    end
+end
+
+# the source invariant is the set union of half-spaces, the other sets are polyhedral
+function _apply(tr::DiscreteTransition{<:Universe, <:ZeroSet, GT, IT⁻, IT⁺},
+                X::ConvexHullArray{N, PT},
+                ::Exact) where {N,
+                                PT<:AbstractPolyhedron{N},
+                                GT<:AbstractPolyhedron{N},
+                                IT⁻<:AbstractPolyhedron{N},
+                                IT⁺<:AbstractPolyhedron{N}}
+
+    vlist = [vertices_list(Xi) for Xi in array(X)]
+    vlist_ch = convex_hull(vcat(vlist...)) # TODO pass the backend as an option
+    Xhrep = tohrep(vlist_ch)
+    out = _apply(tr, Xhrep, Exact())
+    return HPolyhedron(out)
+end
+
+# ---------------------------------
+# Intersection method: BoxEnclosure
+# ---------------------------------
+
+function _apply(tr::DiscreteTransition{<:Universe, <:ZeroSet, GT, IT⁻, IT⁺},
+                X::ConvexHullArray{N, PT},
+                ::BoxEnclosure) where {N,
+                                       PT<:AbstractPolyhedron{N},
+                                       GT<:AbstractPolyhedron{N},
+                                       VN,
+                                       IT⁻<:UnionSetArray{N, LinearConstraint{N, VN}},
+                                       IT⁺<:AbstractPolyhedron{N}}
+    Xbox = overapproximate(X, Hyperrectangle)
+    out = _apply(tr, Xbox, Exact())
+    return overapproximate(out, Hyperrectangle)
```

Comment:
Hm, I am not sure about the first box approximation. Why is that needed?

Reply:
hmm i'm also not convinced about this choice. i think the most reasonable thing to do (at least as a fallback implementation) would be to evaluate the support functions of the lazy intersection array along the box directions (as in https://github.com/JuliaReach/ReachabilityAnalysis.jl/issues/172). . we can revisit this later.

Reply:
the benefit of using `Xbox = overapproximate(X, Hyperrectangle)` is that you don't have to make the possibly expensive `constraints_list(X)`.

Reply:
> the benefit of using Xbox = overapproximate(X, Hyperrectangle) is that you don't have to make the possibly expensive constraints_list(X).

True, but that applies to all other methods as well. With this initial overapproximation, the result is generally not the tightest box approximation. I find it rather arbitrary to do it here. It is certainly one possibility to do that but it may have unexpected results.

Reply:
ok. yes, `BoxEnclosure` should return the tightest box enclosure (if possible).